### PR TITLE
Fixing tracking bug

### PR DESCRIPTION
**Pull request by Kalud-Z**

### CODE REVIEW

File: src/app/front-page/dialog/dialog.component.ts [hunk 1]
```diff
@@ -16,7 +16,9 @@ export class DialogComponent implements OnInit {
   ngOnInit() {
   }
 
+
   login() {
+    //@khaled TODO : activate google tracking without cookies , in case the user rejected the cookies.
```
Comment (msallat) [inline src/app/front-page/dialog/dialog.component.ts:21]: @Kalud-Z still a todo?

File: src/app/tracking/tracking.service.ts [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
+    dataLayer: Record<string, any>[];
+  }
+}
 
 declare const ga: Function;
 
-interface WindowWithDataLayer extends Window {
-  dataLayer: Array<any>;
-}
+// interface WindowWithDataLayer extends Window {
```
Comment (msallat) [inline src/app/tracking/tracking.service.ts:14]: @Kalud-Z we could remove the outcommented code?

File: src/app/tracking/tracking.service.ts [hunk 3]
```diff
@@ -20,13 +25,19 @@ export class TrackingService {
...
   public init() {
+    this.ngcCookieContentService.statusChange$.subscribe(
+        (event: NgcStatusChangeEvent) => {
+          if(event.status === "dismiss") {
+            this.enableGoogleAnalyticsWithoutCookies(this.gaProperty);
+          }
+        });
+
     if (isPlatformBrowser(this.platformId)) {
-      console.log(document.cookie);
```
Comment (msallat) [inline src/app/tracking/tracking.service.ts:29]: @Kalud-Z we could remove the logging?

File: src/app/tracking/tracking.service.ts [hunk 4]
```diff
@@ -47,20 +59,94 @@ export class TrackingService {
...
-      }, i[r].l = 1 * new Date().getTime();
-      a = s.createElement(o),
-        m = s.getElementsByTagName(o)[0];
-      a.async = 1;
-      a.src = g;
-      m.parentNode.insertBefore(a, m);
-    })(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');
-    ga('create', this.gaProperty, 'auto', {cookieFlags: this.flagsValue});
-    ga('set', 'anonymizeIp', true);
+    // (function(i, s, o, g, r, a, m) {
```
Comment (msallat) [inline src/app/tracking/tracking.service.ts:62]: @Kalud-Z we could remove the outcommented code?

File: src/app/front-page/dialog/dialog.component.ts (hunk 1 above)
Comment (Kalud-Z) [reply src/app/front-page/dialog/dialog.component.ts:21]: no, not really. I will delete it, and remove also all unnecessary comments.

**Comment by msallat** [comment]
nice work :)

**Pull request merged**